packages/components/src/components/combo-box/combo-box.tsx

```diff
@@ -122,24 +123,72 @@ export class ComboBox {
     isMouseOverEventDisabled = false;
 
     filterElement!: HTMLInputElement;
+    svgButtonToggleDropdown!: SVGElement;
     clearButtonElement!: HTMLButtonElement;
     listElement!: HTMLElement;
 
     currentItemIndex = 0;
     selectedItemIndex = 0;
     highlightedItemClassName = 'list__item--highlighted';
 
+    /**
+     * If we click on the map (canvas) we want to empty selection, so this.open can be set to false.
+     *
+     * @param {ev} ev
+     */
+    @Listen('mousedown', { target: 'window' })
+    clearSelectionOnMouseDown(ev) {
+        if (!this.hostElement.contains(ev.target) && ev.target !== this.svgButtonToggleDropdown) {
+            window.getSelection().empty();
+        }
+    }
+
     /**
      * Outside the dropdown listener. It will close the dropdown when a click is outside a dropdown and dropdown list.
      *
      * @param {ev} ev
      */
     @Listen('click', { target: 'window' })
+    @Listen('mouseup', { target: 'window' })
     checkForClickOutside(ev) {
-        if (!this.hostElement.contains(ev.target) && this.noResultsMessage) {
-            this.clearFilter();
-            this.filterElement.value = this.selected[0]?.text;
-            this.open = false;
+        // Check if the event target is inside the input field or canvas
+        const clicksInsideBody = document.getElementsByTagName('body');
```

Comment:
I don't think I understand the naming of this variable. `getElementsByTagName('body')` will return a collection of HTML elements (and how can multiple body elements exist in the document? Can't `document.body` be used instead).

How does that relate to something with clicks?